app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java

```diff
@@ -702,31 +704,48 @@ private ActionExecutionResult addDataTypesAndSetSuggestedWidget(ActionExecutionR
      * Suggest the best widget to the query response. We currently planning to support List, Select, Table and Chart widgets
      * @return
      */
-    private WidgetType getSuggestedWidget(Object data) {
+    private List<WidgetType> getSuggestedWidget(Object data) {
 
-        if(data instanceof String) {
-            return WidgetType.TEXT_WIDGET;
-        }
+        List<WidgetType> widgetTypeList = new ArrayList<>();
 
-        if(data instanceof ArrayNode && !((ArrayNode) data).isEmpty()  && ((ArrayNode) data).isArray()) {
-            ArrayNode array = (ArrayNode) data;
-            Integer length = array.size();
-            ObjectNode objectNode = (ObjectNode)array.get(0);
-            Integer fieldsCount = array.get(0).size();
-            if( (objectNode.has("x") || (objectNode.has("X")) )&&
-                    (objectNode.has("y") || (objectNode.has("Y"))) ) {
-                return WidgetType.CHART_WIDGET;
-            }
-            if(fieldsCount <= 2) {
-                return WidgetType.DROP_DOWN_WIDGET;
-            }
-            if(length <= 20 && fieldsCount <= 5) {
-                return WidgetType.LIST_WIDGET;
+        if(data instanceof ArrayNode && ((ArrayNode) data).isArray()) {
+            if(!((ArrayNode) data).isEmpty()) {
+                try {
+                    ArrayNode array = (ArrayNode) data;
+                    int length = array.size();
+                    JsonNode node = array.get(0);
+                    JsonNodeType nodeType = node.getNodeType();
+
+                    if(nodeType.equals(JsonNodeType.STRING)) {
```

Comment:
Nit: Always have the constant value first in an if condition. This ensures that we are protected against NPEs. Goes a long way in ensuring small bugs don't arise across the code base. This should be done across all if condition comparisons in this file.

```suggestion
                    if(JsonNodeType.STRING.equals(nodeType)) {
```

Reply:
Good point, Never thought of this. I will follow this in future :)